pkg/services/baremetal/baremetal/baremetal.go

```diff
@@ -306,6 +312,12 @@ func (s *Service) associate(ctx context.Context) error {
 	}
 
 	s.ensureMachineAnnotation(host)
+
+	s.scope.BareMetalMachine.Status.Phase = clusterv1.MachinePhaseProvisioning
+	if host.Spec.Status.ProvisioningState == infrav1.StateProvisioned {
+		s.scope.BareMetalMachine.Status.Phase = clusterv1.MachinePhaseRunning
```

Comment:
If we make the phase `running` here do we need to make it again running in the reconcile function above?

Reply:
the only place where we decide whether a host is ready or not is in setProviderID.. There you can decide whether something is running or not https://github.com/syself/cluster-api-provider-hetzner/blob/ani%2Fissues%2F941/pkg/services/baremetal/baremetal/baremetal.go#L549